### DISCUSSION TITLE
Could we have more different color for syntax highlighting?

### DISCUSSION THREAD

**Discussion by pmattei** [ocamllabs/vscode-ocaml-platform#929] [Ideas]
Hello

Would it be possible to have some syntax highlighting customization options or an explanation of how to change that?
In particular I would like that my module names have different color than the language keywords.
Enumerated types are also not language keywords so should have different color for me
The code structures could be in different colors, in order to identify the beginning and the end at a glance.
(I think about begin / end or match and |)

Thank

Regards

**Comment by mlantas** [comment]
> In particular I would like that my module names have different color than the language keywords.

This is only a problem with the default VSCode theme. The extension doesn't define any colors, only token names.

If you would like your syntax highlighting to use different colors, you need to use a different theme. Either choose one of the other preinstalled themes or install a theme from the marketplace: https://marketplace.visualstudio.com/search?target=VSCode&category=Themes&sortBy=Installs

See also this comment: https://github.com/ocamllabs/vscode-ocaml-platform/issues/562#issuecomment-797560573

> The code structures could be in different colors, in order to identify the beginning and the end at a glance.

The syntax isn't currently specific enough for this. Both `begin` and `match` are recognized as `keyword.other.ocaml`: 
https://github.com/ocamllabs/vscode-ocaml-platform/blob/35903d7bb7032ad5a23606dc91b03905788146c0/syntaxes/ocaml.json#L501-L505

If `begin` and `match` had different token names, it would still require you to either find or make a theme yourself that supports different colors per language construct.

**Comment by pmattei** [comment]
Thank you for your reply
I thought Ocaml platform had defined the highlighting
I didn't think it was the default of VS Code, sorry
